platform_plugin_aspects/xblock.py

```diff
@@ -0,0 +1,175 @@
+"""XBlock to embed Superset dashboards in Open edX."""
+
+from __future__ import annotations
+
+import logging
+
+import pkg_resources
+from django.utils import translation
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import List, Scope, String
+from xblock.utils.resources import ResourceLoader
+from xblock.utils.studio_editable import StudioEditableXBlockMixin
+
+from .utils import _, generate_superset_context
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+@XBlock.needs("user")
+@XBlock.needs("i18n")
+class SupersetXBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    XBlock provides a way to embed dashboards from Superset in a course.
+    """
+
+    editable_fields = ("display_name", "dashboard_uuid", "filters")
+
+    display_name = String(
+        display_name=_("Display name"),
+        help=_("Display name"),
+        default="Superset Dashboard",
+        scope=Scope.settings,
+    )
+
+    dashboard_uuid = String(
+        display_name=_("Dashboard UUID"),
+        help=_(
+            "The ID of the dashboard to embed. Available in the Superset embed dashboard UI."
+        ),
+        default="",
+        scope=Scope.settings,
+    )
+
+    filters = List(
+        display_name=_("Filters"),
+        help=_(
+            """List of SQL filters to apply to the
+               dashboard. E.g: ["org='edX'", "country in ('us', 'co')"]
+               The fields used here must be available on every dataset used by the dashboard.
+               """
+        ),
+        default=[],
+        scope=Scope.settings,
+    )
+
+    def render_template(self, template_path, context=None) -> str:
+        """
+        Render a template with the given context.
+
+        The template is translatedaccording to the user's language.
+
+        args:
+            template_path: The path to the template
+            context: The context to render in the template
+
+        returns:
+            The rendered template
+        """
+        return loader.render_django_template(
+            template_path, context, i18n_service=self.runtime.service(self, "i18n")
+        )
+
+    def user_is_student(self, user) -> bool:
+        """
+        Check if the user is a student.
+        """
+        return not user or user.opt_attrs.get("edx-platform.user_role") == "student"
+
+    def student_view(self, context=None):
```

Comment:
If this XBlock will only be available for instructors you can use `instructor_view`, however, I think eventually there will be some reports that can be available to students. Just thinking, not a blocker